common/buildcraft/lib/client/guide/parts/recipe/GuideCraftingFactory.java

```diff
@@ -128,15 +128,16 @@ public static GuidePartFactory getFactory(IRecipe recipe) {
             ChangingItemStack[][] inputGrid = { input };
             if (recipe instanceof IRecipeViewable.IViewableGrid) {
                 int width = ((IRecipeViewable.IViewableGrid) recipe).getRecipeWidth();
-                inputGrid = new ChangingItemStack[inputGrid.length / width][width];
+                int height = ((IRecipeViewable.IViewableGrid) recipe).getRecipeHeight();
+                inputGrid = new ChangingItemStack[width][height];
                 int x = 0;
                 int y = 0;
                 for (ChangingItemStack anInput : input) {
                     inputGrid[x][y] = anInput;
-                    x++;
-                    if (x > width) {
-                        x = 0;
-                        y++;
+                    y++;
+                    if (y >= height) {
+                        y = 0;
+                        x++;
```

Comment:
`GuideCrafting#renderIntoArea` looks to be the cause of it working by column instead of row, it consistently iterates each `y` for every `x` rather than the other way around.

As for the recipe crash, there's nothing natively in BuildCraft that causes it at the moment because the only thing that implements `IViewableGrid` is the facade swapping recipes, which are 1x1. That's probably why this hasn't come up before. I was making a version of IC2's `AdvRecipe` that also implemented `IViewableGrid`, but something like [this](https://gist.github.com/Chocohead/911420cf88df2580ac7639ff99e46903) should show what happens.